lib/src/server.dart

```diff
@@ -170,22 +173,21 @@ class Server {
         'result': result,
         'id': request['id']
       };
-    }).catchError((error, stackTrace) {
-      if (error is! RpcException) {
-        error = new RpcException(
-            error_code.SERVER_ERROR, getErrorMessage(error), data: {
-          'full': error.toString(),
-          'stack': new Chain.forTrace(stackTrace).toString()
-        });
-      }
-
+    } catch (error, stackTrace) {
       if (error.code != error_code.INVALID_REQUEST &&
```

Comment:
Good point. What about:

```dart
} catch (error, stackTrace) {
  if (error is RpcException) {
    if (error.code != error_code.INVALID_REQUEST &&
        !request.containsKey("id")) {
      return null;
    } else {
      return error.serialize(request);
    }
  } else if (!request.containsKey("id")) {
    return null;
  }

  var chain = new Chain.forTrace(stackTrace);
  return new RpcException(error_code.SERVER_ERROR, getErrorMessage(error),
          data: {'full': error.toString(), 'stack': chain.toString()})
      .serialize(request);
}
```

There's still a little duplication, but it avoids allocating a `RpcException` just to return `null`.